### PR TITLE
feat(KFLUXBUGS-1540): filter on Applications

### PR DESCRIPTION
**Pull request by caugello**
## Fixes 
[KFLUXBUGS-1540](https://issues.redhat.com//browse/KFLUXBUGS-1540)

## Description
Application list can be filtered by name

## Type of change

- [X] Feature

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

File: src/components/ApplicationListView/ApplicationListView.tsx [hunk 1]
```diff
@@ -36,13 +40,35 @@ const ApplicationListView: React.FC<React.PropsWithChildren<unknown>> = () => {
...
+  }, [nameFilter]);
 
   const [applications, loaded] = useApplications(namespace);
   applications?.sort(
     (app1, app2) =>
       +new Date(app2.metadata.creationTimestamp) - +new Date(app1.metadata.creationTimestamp),
   );
 
+  const filteredApplications = React.useMemo(() => {
+    return applications?.filter((app) => app.metadata.name.includes(nameFilter));
```
Comment (sahil143) [inline src/components/ApplicationListView/ApplicationListView.tsx:59]: If `app.spec.displayName` is available, filter list based on that else fallback on `app.metadata.name`.

Comment (caugello) [reply src/components/ApplicationListView/ApplicationListView.tsx]: Added fallback and test

**Review by sahil143 — CHANGES_REQUESTED** [review]
@caugello Filter doesn't seem to be working for case-sensitive terms. Maybe we need to convert both nameFilter and displayName to lower case before comparing.

image
image

File: src/components/ApplicationListView/ApplicationListView.tsx [hunk 2]
```diff
@@ -36,13 +40,40 @@ const ApplicationListView: React.FC<React.PropsWithChildren<unknown>> = () => {
   const applicationBreadcrumbs = useApplicationBreadcrumbs();
   const [canCreateApplication] = useAccessReviewForModel(ApplicationModel, 'create');
   const [canCreateComponent] = useAccessReviewForModel(ComponentModel, 'create');
+  const [nameFilter, setNameFilter] = useSearchParam('name', '');
+  const [onLoadName, setOnLoadName] = React.useState(nameFilter);
+
+  React.useEffect(() => {
```
Comment (abhinandan13jan) [inline src/components/ApplicationListView/ApplicationListView.tsx:46]: Why do we need a `nameFilter` and also an `onLoadName` , in this hook, the `onLoadName` keeps updating with the `nameFilter` doesn't it?

Comment (caugello) [reply src/components/ApplicationListView/ApplicationListView.tsx:46]: Seems redundant, we can only use the query params from namefilter. Removed it.

**Comment by caugello** [comment]
@sahil143 @abhinandan13jan anything blocking to merge this ?

**Pull request merged**